### PR TITLE
chore(profiling): standardize class name propagation

### PR DESCRIPTION
**Pull request by sanchda**
Right now, class names are only propagated on the leaf frame of a stack, and only then as a label.  This is unlike our other profilers and is also impeding progress on related products.

This PR changes the names so they are sent as `class_name`.`function_name`, which is slightly more consistent.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [X] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: releasenotes/notes/profiling-send-classnames-3a65dc5b33be5b0a.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    profiling: adds classnames to each frames, when possible
```
Comment (emmettbutler) [inline releasenotes/notes/profiling-send-classnames-3a65dc5b33be5b0a.yaml:4]:
```suggestion
    profiling: adds the class name to each frame, when possible
```

**Review by emmettbutler — CHANGES_REQUESTED** [review]

File: releasenotes/notes/profiling-send-classnames-3a65dc5b33be5b0a.yaml (hunk 1 above)
Comment (sanchda) [reply releasenotes/notes/profiling-send-classnames-3a65dc5b33be5b0a.yaml]: @emmettbutler went ahead and took this, thanks!

**Review by emmettbutler — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

File: ddtrace/profiling/collector/_traceback.pyx [hunk 2]
```diff
@@ -45,7 +45,14 @@ cpdef traceback_to_frames(traceback, max_nframes):
             frame = tb.tb_frame
             code = frame.f_code
             lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, DDFrame(code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+
+            # If the class_name is nonempty, then prepend the classname to the function name like class.function
+            # We'll ignoer the old placeholder for the class_name for now
+            class_name = _extract_class_name(frame)
+            if class_name:
+                frames.append(DDFrame(code.co_filename, lineno, class_name + "." + code.co_name))
```
Comment (P403n1x87) [inline ddtrace/profiling/collector/_traceback.pyx:53]: Starting with CPython 3.11 we have `co_qualname` that we can use

File: ddtrace/profiling/collector/_traceback.pyx [hunk 3]
```diff
@@ -45,7 +45,14 @@ cpdef traceback_to_frames(traceback, max_nframes):
             frame = tb.tb_frame
             code = frame.f_code
             lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, DDFrame(code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+
+            # If the class_name is nonempty, then prepend the classname to the function name like class.function
+            # We'll ignoer the old placeholder for the class_name for now
```
Comment (P403n1x87) [inline ddtrace/profiling/collector/_traceback.pyx:50]:
```suggestion
            # We'll ignore the old placeholder for the class_name for now
```

File: ddtrace/profiling/collector/_traceback.pyx (hunk 2 above)
Comment (sanchda) [reply ddtrace/profiling/collector/_traceback.pyx]: Good stuff, added support in ef884ae68

File: ddtrace/profiling/collector/_traceback.pyx (hunk 3 above)
Comment (sanchda) [reply ddtrace/profiling/collector/_traceback.pyx]: Thank you for catching that!

**Comment by sanchda** [comment]
:wave: I messed up a rebase and pulled in a lot of you nice folks unnecessarily.  To make life easy for the many, I'm going to close this PR and try again.  Apologies for the unnecessary noise, everyone!  My fault for trying to keep too many plates spinning.

**Comment by albertvaka** [comment]
New PR: https://github.com/DataDog/dd-trace-py/pull/7620

**Pull request closed without merging**